### PR TITLE
 Replace collections helper with modifying objects directly

### PR DESCRIPTION
**Pull request by disrani-px**
**What type of PR is this?**
Cleanup

**What this PR does / why we need it**:
The vendor update fixes issue when api services are down and we try to discover object types

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:

```release-note
Fixed issue with migration controller trying to fetch API types from an API service which was offline
```

**Does this change need to be cherry-picked to a release branch?**:

No

### CODE REVIEW

**Comment by disrani-px** [comment]
Going to run through the integration tests before merging

**Comment by disrani-px** [comment]
All the tests passed
```
04:53:48.456 --- SKIP: TestMain (15208.32s)
04:53:48.456     --- PASS: TestMain/setup (21.62s)
04:53:48.456     --- PASS: TestMain/Extender (1220.75s)
04:53:48.456         --- PASS: TestMain/Extender/pvcOwnershipTest (400.15s)
04:53:48.456         --- PASS: TestMain/Extender/noPVCTest (60.12s)
04:53:48.456         --- PASS: TestMain/Extender/singlePVCTest (90.77s)
04:53:48.456         --- PASS: TestMain/Extender/statefulsetTest (200.45s)
04:53:48.456         --- PASS: TestMain/Extender/multiplePVCTest (60.50s)
04:53:48.456         --- PASS: TestMain/Extender/driverNodeErrorTest (408.75s)
04:53:48.456     --- PASS: TestMain/HealthMonitor (983.20s)
04:53:48.461         --- PASS: TestMain/HealthMonitor/stopDriverTest (452.42s)
04:53:48.462         --- PASS: TestMain/HealthMonitor/stopKubeletTest (530.78s)
04:53:48.462     --- PASS: TestMain/Snapshot (3504.27s)
04:53:48.464         --- PASS: TestMain/Snapshot/simpleSnapshotTest (70.79s)
04:53:48.464         --- PASS: TestMain/Snapshot/cloudSnapshotTest (100.32s)
04:53:48.464         --- PASS: TestMain/Snapshot/snapshotScaleTest (244.41s)
04:53:48.464         --- PASS: TestMain/Snapshot/groupSnapshotTest (583.19s)
04:53:48.466         --- PASS: TestMain/Snapshot/groupSnapshotScaleTest (446.21s)
04:53:48.466         --- PASS: TestMain/Snapshot/scheduleTests (2014.24s)
04:53:48.466             --- PASS: TestMain/Snapshot/scheduleTests/intervalTest (470.31s)
04:53:48.466             --- PASS: TestMain/Snapshot/scheduleTests/dailyTest (396.65s)
04:53:48.466             --- PASS: TestMain/Snapshot/scheduleTests/weeklyTest (426.63s)
04:53:48.466             --- PASS: TestMain/Snapshot/scheduleTests/monthlyTest (426.94s)
04:53:48.468             --- PASS: TestMain/Snapshot/scheduleTests/invalidPolicyTest (290.31s)
04:53:48.468         --- PASS: TestMain/Snapshot/storageclassTests (45.10s)
04:53:48.468     --- PASS: TestMain/SnapshotRestore (132.39s)
04:53:48.468         --- PASS: TestMain/SnapshotRestore/simpleSnapshotRestoreTest (71.97s)
04:53:48.468         --- PASS: TestMain/SnapshotRestore/groupSnapshotRestoreTest (60.42s)
04:53:48.468         --- SKIP: TestMain/SnapshotRestore/cloudSnapshotRestoreTest (0.00s)
04:53:48.468             snapshot_restore_test.go:74: Skipping CloudSnapRestore integration tests
04:53:48.468         --- PASS: TestMain/SnapshotRestore/groupCloudSnapshotRestoreTest (0.00s)
04:53:48.471     --- PASS: TestMain/CmdExecutor (57.72s)
04:53:48.471     --- PASS: TestMain/Migration (3972.37s)
04:53:48.471         --- PASS: TestMain/Migration/deploymentTest (240.80s)
04:53:48.471         --- PASS: TestMain/Migration/statefulsetTest (331.19s)
04:53:48.471         --- PASS: TestMain/Migration/statefulsetRuleTest (351.16s)
04:53:48.471         --- PASS: TestMain/Migration/preExecRuleMissingTest (250.46s)
04:53:48.471         --- PASS: TestMain/Migration/postExecRuleMissingTest (240.82s)
04:53:48.471         --- PASS: TestMain/Migration/disallowedNamespaceTest (250.43s)
04:53:48.471         --- PASS: TestMain/Migration/failingPreExecRuleTest (250.42s)
04:53:48.471         --- PASS: TestMain/Migration/failingPostExecRuleTest (170.50s)
04:53:48.471         --- PASS: TestMain/Migration/labelSelectorTest (541.19s)
04:53:48.475         --- PASS: TestMain/Migration/intervalScheduleTest (193.61s)
04:53:48.475         --- PASS: TestMain/Migration/dailyScheduleTest (312.78s)
04:53:48.477         --- PASS: TestMain/Migration/weeklyScheduleTest (282.79s)
04:53:48.477         --- PASS: TestMain/Migration/monthlyScheduleTest (362.70s)
04:53:48.477         --- PASS: TestMain/Migration/scheduleInvalidTest (190.52s)
04:53:48.478     --- PASS: TestMain/ApplicationClone (1829.37s)
04:53:48.478         --- PASS: TestMain/ApplicationClone/deploymentTest (190.41s)
04:53:48.478         --- PASS: TestMain/ApplicationClone/statefulsetTest (240.59s)
04:53:48.478         --- PASS: TestMain/ApplicationClone/statefulsetRuleTest (260.73s)
04:53:48.478         --- PASS: TestMain/ApplicationClone/preExecRuleMissingTest (165.30s)
04:53:48.478         --- PASS: TestMain/ApplicationClone/postExecRuleMissingTest (175.32s)
04:53:48.481         --- PASS: TestMain/ApplicationClone/disallowedNamespaceTest (155.26s)
04:53:48.481         --- PASS: TestMain/ApplicationClone/failingPreExecRuleTest (175.29s)
04:53:48.481         --- PASS: TestMain/ApplicationClone/failingPostExecRuleTest (155.25s)
04:53:48.481         --- PASS: TestMain/ApplicationClone/labelSelectorTest (311.23s)
04:53:48.481     --- PASS: TestMain/ApplicationBackup (3486.63s)
04:53:48.481         --- PASS: TestMain/ApplicationBackup/applicationBackupRestoreTest (230.59s)
04:53:48.481         --- PASS: TestMain/ApplicationBackup/preExecRuleTest (260.63s)
04:53:48.481         --- PASS: TestMain/ApplicationBackup/postExecRuleTest (290.60s)
04:53:48.481         --- PASS: TestMain/ApplicationBackup/preExecMissingRuleTest (140.30s)
04:53:48.481         --- PASS: TestMain/ApplicationBackup/postExecMissingRuleTest (150.29s)
04:53:48.481         --- PASS: TestMain/ApplicationBackup/preExecFailingRuleTest (140.67s)
04:53:48.481         --- PASS: TestMain/ApplicationBackup/postExecFailingRuleTest (160.34s)
04:53:48.481         --- PASS: TestMain/ApplicationBackup/scheduleTests (2113.22s)
04:53:48.481             --- PASS: TestMain/ApplicationBackup/scheduleTests/intervalTest (480.39s)
04:53:48.481             --- PASS: TestMain/ApplicationBackup/scheduleTests/dailyTest (456.55s)
04:53:48.481             --- PASS: TestMain/ApplicationBackup/scheduleTests/weeklyTest (416.48s)
04:53:48.481             --- PASS: TestMain/ApplicationBackup/scheduleTests/monthlyTest (476.49s)
04:53:48.481             --- PASS: TestMain/ApplicationBackup/scheduleTests/invalidPolicyTest (280.30s)
04:53:48.481     common_test.go:118: Skipping cluster domain tests
04:53:48.481 PASS
04:53:48.488 
04:53:48.488 === Skipped
04:53:48.490 === SKIP: . TestMain/SnapshotRestore/cloudSnapshotRestoreTest (0.00s)
04:53:48.490         --- SKIP: TestMain/SnapshotRestore/cloudSnapshotRestoreTest (0.00s)
04:53:48.490             snapshot_restore_test.go:74: Skipping CloudSnapRestore integration tests
04:53:48.491 
04:53:48.491 === SKIP: . TestMain (15208.32s)
```

File: pkg/resourcecollector/persistentvolume.go [hunk 1]
```diff
@@ -16,28 +16,22 @@ func (r *ResourceCollector) pvToBeCollected(
...
+		return false, fmt.Errorf("error converting to persistent volume: %v", err)
 	}
+
 	// Only collect Bound PVs
-	if phase != string(v1.ClaimBound) {
+	if pv.Status.Phase != v1.VolumeBound {
 		return false, nil
 	}
 
+	pvcName := pv.Spec.ClaimRef.Name
```
Comment (ram-infrac) [inline pkg/resourcecollector/persistentvolume.go]: I guess,  need to add nil check for `ClaimRef`

Comment (disrani-px) [reply pkg/resourcecollector/persistentvolume.go]: Fixed

**Pull request merged**